cmd/charm/charmcmd/plugin.go

```diff
@@ -114,53 +114,107 @@ var pluginDescriptionsResults []pluginDescription
 // getPluginDescriptions runs each plugin with "--description".  The calls to
 // the plugins are run in parallel, so the function should only take as long
 // as the longest call.
+// But there was a bug reported on this behavior and it is noticably slow
+// so cache the results and check each plugins mtime to invalidate the plugin.
 func getPluginDescriptions() []pluginDescription {
```

Comment:
This function is now over 100 lines. That is not reasonable, it should be split into smaller logical parts.